### PR TITLE
Gh 134

### PR DESCRIPTION
**Pull request by R-Lawton**
# Verification
## Multi-cluster
1. run `make local-setup MCTC_WORKLOAD_CLUSTERS_COUNT=2`
2. run the addon manager `make run-ocm`
3. Apply the RBAC and ManagedClusterAddOn and ClusterManagementAddOn `kubectl apply -f config/kuadrant-addon/deploy/hub` Note please replace the cluster-name with the spoke cluster you want to add to 
4. A kuadrant-operator subscription and namespace called kuadrant-system and kuadrant cr should be created in each spoke cluster
5. The kuadrant operator should be created in the operators namespace
6. Check the health prober via `kubectl get ManagedClusterAddOn -n kind-mctc-workload-{number}` available should be True meaning the subscription catalogue source is in a healthy state reporting back from each spoke cluster
## Single
1. run `make local-setup OCM_SINGLE=true`
2. run the addon manager `make run-ocm`
3. Apply the RBAC and ManagedClusterAddOn and ClusterManagementAddOn `kubectl apply -f config/kuadrant-addon/deploy/hub` Note please replace the cluster-name with the spoke cluster you want to add to 
4. A kuadrant-operator subscription and namespace called kuadrant-system and kuadrant cr should be created in each spoke cluster
5. The kuadrant operator should be created in the operators namespace
6. Check the health prober via `kubectl get ManagedClusterAddOn -n kind-mctc-workload-{number}` available should be True meaning the subscription catalogue source is in a healthy state reporting back from each spoke cluster

### CODE REVIEW

File: cmd/ocm/addon-manager/manifests/subscription.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: kuadrant-operator
+  namespace: operators
```
Comment (maleck13) [inline cmd/ocm/addon-manager/manifests/subscription.yaml:5]: is it this ns that is causing the install in the wrong place? what if this was set to kuadrant-system?

File: pkg/controllers/gateway/gateway_controller.go [hunk 2]
```diff
@@ -55,7 +55,10 @@ const (
 	ClusterSyncerAnnotation               = "clustersync.kuadrant.io"
 	GatewayClusterLabelSelectorAnnotation = "kuadrant.io/gateway-cluster-label-selector"
 	GatewayFinalizer                      = "kuadrant.io/gateway-cleanup"
+<<<<<<< HEAD
```
Comment (maleck13) [inline pkg/controllers/gateway/gateway_controller.go:58]: merge conflict

**Review by maleck13 — COMMENTED** [review]
looks like there is a merge conflict still present

File: cmd/ocm/addon-manager/manifests/subscription.yaml (hunk 1 above)
Comment (R-Lawton) [reply cmd/ocm/addon-manager/manifests/subscription.yaml]: No i was misunderstanding how operator groups worked this is working now

File: Dockerfile [hunk 3]
```diff
@@ -54,4 +58,13 @@ WORKDIR /
 COPY --from=syncer_builder /workspace/syncer .
 USER 65532:65532
 
-ENTRYPOINT ["/syncer"]
\ No newline at end of file
+ENTRYPOINT ["/syncer"]
```
Comment (maleck13) [inline Dockerfile:61]: I think we can get rid of this

File: config/kuadrant-addon/deploy/hub/managed-cluster-addon.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedClusterAddOn
+metadata:
+ name: kuadrant-addon
+ namespace: <cluster_namespace>
+spec:
+ installNamespace: open-cluster-management-agent-addon
+ 
+ 
+
```
Comment (maleck13) [inline config/kuadrant-addon/deploy/hub/managed-cluster-addon.yaml:10]: prob remove white space here

File: pkg/ocm/hub/addon.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+						Name:      "kuadrant-operator",
+						Namespace: "operators",
+					},
+					ProbeRules: []workapiv1.FeedbackRule{
+						{
+							Type: workapiv1.JSONPathsType,
+							JsonPaths: []workapiv1.JsonPath{
+								{
+									Name:    "healthy",
+									Path:    ".status.catalogHealth[0].healthy",
```
Comment (maleck13) [inline pkg/ocm/hub/addon.go:29]: is it always the 0 index?

**Review by maleck13 — APPROVED** [review]
generally looking ok. couple of very small things. Will try out locally and then merge

**Pull request merged**